src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+brand new HTML documents, and calling `skipWaiting()` ensures that an
+updated service worker will get a chance to handle those new navigation requests
+immediately after installation. Similarly, calling `clients.claim()`
+means that the updated service worker will get a chance to start controlling any
+open Google Search pages that are uncontrolled, following service worker
+activation.
+
+The approach that Google Search went with isn't necessarily a solution that will
+work for everyone—it was the result of carefully A/B testing various
+combinations of serving options until they found what worked best for them.
+Developers whose backend infrastructure allow them to deploy updates more
+quickly might prefer that the browser check for an updated service worker script
+as frequently as possible, by
+[always ignoring the HTTP cache](https://developers.google.com/web/updates/2018/06/fresher-sw#whats_changing).
+If you're building a single page app that users will might keep open for a long
+period of time, using `skipWaiting()` is probably not the right choice for
+you—you
+[risk running into cache inconsistencies](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#skip_the_waiting_phase)
+if you allow the new service worker to activate while there are long-lived
+clients.
+
+## Key Takeaways
+
+### By default, service workers aren't performance neutral
+
+Adding a service worker to your web app means inserting an additional piece of
+JavaScript that needs to be loaded and executed before your web app gets
+responses to its requests. If those responses end up coming from a local cache
+rather than from the network, then the overhead of running the service worker
+will usually be negligible in comparison to the performance win from going
+cache-first. But if you know that your service worker will always have to
+consult the network when handling navigation requests, using navigation preload
+is a crucial performance win.
+
+### Service workers are (still!) a progressive enhancement
+
+The service worker support story is much brighter today than it was even a year
+ago. All modern browsers now feature at least some
+[support for service workers](https://jakearchibald.github.io/isserviceworkerready/),
+but unfortunately, there are some advanced service worker features—like
+background sync and navigation preload—that aren't rolled out universally.
+Feature checking for the specific subset of features that you know you need, and
+only registering a service worker when those are present, is still a reasonable
+approach to take.
+
+Similarly, if you've run experiments in the wild, and know that low-end devices
+end up performing poorly with the additional overhead of a service worker, you
+can abstain from registering a service worker in those scenarios as well.
+
+You should continue to treat service workers as a [progressive enhancement](https://en.wikipedia.org/wiki/Progressive_enhancement)
```

Comment:
More opportunities to remove 'will' and clean up a little:

'... that gets added to a web app when all the prerequisites are met and the service worker adds something positive to user experience and overall loading performance.